examples/train_val_test_split_example.ipynb

```diff
@@ -0,0 +1,195 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "6b73362c",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0.3515039651924021"
+      ]
+     },
+     "execution_count": 26,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from sklearn.ensemble import RandomForestRegressor\n",
+    "rfr_baseline = RandomForestRegressor(n_estimators=5)\n",
+    "rfr_baseline.fit(X_train, y_train)\n",
+    "rfr_baseline.score(X_val, y_val)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb00d17a",
+   "metadata": {},
+   "source": [
+    "Now try and find some better model parameters by tuning the model, in this case with an automatic tuner:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "a3258b4e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Fitting 3 folds for each of 50 candidates, totalling 150 fits\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "0.5287961605324687"
+      ]
+     },
+     "execution_count": 28,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from sklearn.model_selection import RandomizedSearchCV\n",
+    "import numpy as np\n",
+    "\n",
+    "rdr_tuned = RandomForestRegressor()\n",
+    "\n",
+    "n_estimators = np.arange(5, 50, step=5)\n",
+    "max_depth = list(np.arange(2, 20, step=2)) + [None]\n",
+    "\n",
+    "param_grid = {\n",
+    "    \"n_estimators\": n_estimators,\n",
+    "    \"max_depth\": max_depth,\n",
+    "}\n",
+    "\n",
+    "random_cv = RandomizedSearchCV(\n",
+    "    rdr_tuned, param_grid, cv=3, n_iter=50, scoring=\"r2\", n_jobs=-1, verbose=1\n",
+    ")\n",
+    "random_cv.fit(X_train, y_train)\n",
+    "random_cv.score(X_val, y_val)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5fc16d62",
+   "metadata": {},
+   "source": [
+    "Before celebrating our substantial improvement, let's make sure that the model performs well on the data it was not tuned to be good on:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "1cad8cd4",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0.47114420960992875"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "random_cv.score(X_test, y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6ff82d2e",
+   "metadata": {},
+   "source": [
+    "The performance is lower on the out-of-sample data than it is on the tuning data, which is a sign that the results might not be generalizable. If future measurements were to be taken, we cannot be sure that they would work with this data. We should try to improve our model further or re-evaluate our modeling approach!"
```

Comment:
If I understand, I believe the validation data is not used during the hyper parameter search. Instead, `RandomizedSearchCV` takes the passed training data and creates 3 different train:val folds of size 67%:33% to do the search. Then because `refit=True` by default, it will retrain a model with the best hyperparameters on the training set which is what makes this comparable to the cells above. So I believe the higher performance on X_val is spurious, and it isn't correct to conclude this is an expected result since I don't believe X_val was used during the hyperparameter search.

Reply:
I'm kind of using the automatic hyperparameter tuning as a stand in for actual human work (couldn't think of a better way to do it in a NB without just saying "and here's what the parameters would be if you wanted to make it predict better based on hand-tuning"). The point I was trying to make is that the user likely did some tuning on their model to make it perform really well on the validation data, but what they should have done was then subject it to _another_ set to see how it performs there as well.

I have added some comments to this effect in the NB -- let me know if they make sense.

I think the only way to use the validation and training data in the automated search would be to pass in both to `fit` but I didn't do that because I don't want people to think that's a good idea. Our point is that tuning your model to perform well on a set will of course make it really good on that set (whether tuning by hand or with this automatic tool), and that you should instead keep some data out.